File: schemas/event-management-ticketing/convex/event_organizers.ts
// event_organizers: users assigned to manage or staff an event with specific roles.
// See README.md for full design rationale.

import { defineTable } from "convex/server";
import { v } from "convex/values";

export const event_organizers = defineTable({
  eventId: v.id("events"),
  userId: v.id("users"),
  role: v.union(
    v.literal("owner"),
    v.literal("admin"),
    v.literal("moderator"),
    v.literal("check_in_staff")
  ),
})
  .index("by_event_id_user_id", ["eventId", "userId"])
  .index("by_user_id", ["userId"]);

File: schemas/event-management-ticketing/spacetimedb/event_organizers.rs
// event_organizers: Users assigned to manage or staff an event.
// See README.md for full design rationale.

use spacetimedb::{SpacetimeType, Timestamp};

#[derive(SpacetimeType, Clone)]
pub enum OrganizerRole {
    Owner, // type: String
    Admin,
    Moderator,
    CheckInStaff,
}

#[spacetimedb::table(name = event_organizers, public)]
pub struct EventOrganizer {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub event_id: String, // UUID — FK → events.id (cascade delete)

    #[index(btree)]
    pub user_id: String, // UUID — FK → users.id (cascade delete)

    pub role: OrganizerRole,
    pub created_at: Timestamp,
    // Composite unique: (event_id, user_id)
}
